packages/ckeditor5-table/src/tablelayout/tablelayoutediting.ts

```diff
@@ -161,6 +164,26 @@ export default class TableLayoutEditing extends Plugin {
 						}
 					}
 				}
+
+				// Remove disallowed attributes and children for layout tables
+				// when `tableType` attribute has been changed by `TableTypeCommand`.
+				if ( entry.type == 'attribute' && entry.attributeKey == 'tableType' ) {
+					for ( const item of entry.range.getItems() ) {
+						if ( item.is( 'element', 'table' ) ) {
+							editor.model.schema.removeDisallowedAttributes( [ item ], writer );
+
+							const tableChildren = item.getChildren();
+
+							// Check if all children are allowed for the new table type.
+							for ( const child of tableChildren ) {
+								if ( !editor.model.schema.checkChild( item, child ) ) {
+									writer.remove( child );
+									hasChanged = true;
+								}
+							}
+						}
+					}
+				}
```

Comment:
What about some captions added to a layout table? Or a heading cols/rows set on it?

Reply:
Ok, let's leave it for a potential follow-up if such a case appears.